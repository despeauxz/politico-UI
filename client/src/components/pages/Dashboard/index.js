import React, { Component, Fragment } from 'react';
import SideNav from '@containers/shared/SideNav';
import Admin from '@containers/shared/Admin';
import Home from '@containers/shared/Home';
import AuthHeader from '@components/shared/Header/AuthHeader';
import { userPropType } from '@helpers/proptypes';


class Dashboard extends Component {
    constructor(props) {
        super(props);

    }

    static propTypes = {
        ...userPropType
    }

    getPageRender = () => {
        switch (this.props.user.isAdmin) {
        case true:
            return <Admin />;
        default:
            return <Home />;
        }
    }


    render() {
        const page = this.getPageRender();

        return (
            <Fragment>
                <div className="font-sans antialiased h-screen flex">
                    <SideNav />
                    <div className="flex-1 flex flex-col bg-white overflow-hidden">
                        <AuthHeader />

                        <div className="px-6 py-4 flex-1 overflow-y-scroll">
                            { page }
                        </div>
                    </div>
                </div>
            </Fragment>
        );
    }
}


export default Dashboard;
