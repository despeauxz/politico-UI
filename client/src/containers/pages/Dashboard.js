import { connect } from 'react-redux';
import Dashboard from '@components/pages/Dashboard';

const mapStateToProps = state => ({
    isAuthenticated: state.auth.isAuthenticated,
    user: state.auth.user
});

export default connect(mapStateToProps)(Dashboard);