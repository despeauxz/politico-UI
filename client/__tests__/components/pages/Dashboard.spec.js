import React from 'react';
import toJson from 'enzyme-to-json';
import Dashboard from '@components/pages/Dashboard';

const Adminprops = {
    isAuthenticated: true,
    user: {
        isAdmin: true
    }
};

const Userprops = {
    isAuthenticated: true,
    user: {
        isAdmin: null
    }
};

describe('<Dashboard/>', () => {
    it('renders Admin Dashboard page component without crashing', () => {
        const shallowWrapper = shallow(<Dashboard { ...Adminprops } />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Admin')).toBeTruthy();

    });

    it('renders Home page component without crashing', () => {
        const shallowWrapper = shallow(<Dashboard { ...Userprops } />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Home')).toBeTruthy();
    });
});