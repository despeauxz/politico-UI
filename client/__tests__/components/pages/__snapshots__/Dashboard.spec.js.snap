// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> renders Home page component without crashing 1`] = `
<Fragment>
  <Connect(HomeHeader) />
  <Sidebar />
  <div
    className="container mx-auto"
  />
</Fragment>
`;
